src/Algolia.Search/Serializer/ConsequenceQueryConverter.cs

```diff
@@ -0,0 +1,124 @@
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*/
+
+using Algolia.Search.Models.Enums;
+using Algolia.Search.Models.Rules;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Algolia.Search.Serializer
+{
+    /// <summary>
+    /// Custom converter for legacy EDIT objects
+    /// </summary>
+    public class ConsequenceQueryConverter : JsonConverter<ConsequenceQuery>
+    {
+        /// <summary>
+        /// Converter for handling legacy Edit
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="objectType"></param>
+        /// <param name="existingValue"></param>
+        /// <param name="hasExistingValue"></param>
+        /// <param name="serializer"></param>
+        /// <returns></returns>
+        public override ConsequenceQuery ReadJson(JsonReader reader, Type objectType, ConsequenceQuery existingValue,
+            bool hasExistingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.Null)
+                return null;
+            if (reader.TokenType != JsonToken.StartObject)
+                return null;
+
+            ConsequenceQuery ret = new ConsequenceQuery();
+
+            var tokens = JToken.Load(reader);
+            foreach (var token in tokens)
+            {
+                var isProperty = token.Type == JTokenType.Property;
+                if (!isProperty)
+                {
+                    continue;
+                }
+
+                var property = (JProperty)token;
+                var isArray = property.Value.Type == JTokenType.Array;
+                if (!isArray)
+                {
+                    continue;
+                }
+
+                var array = (JArray)property.Value;
+                List<Edit> edits = new List<Edit>();
+
+                switch (property.Name)
+                {
+                    case "remove":
+                        List<string> removeList = array.ToObject<List<string>>();
+                        if (removeList.Count > 0)
+                        {
+                            foreach (var r in removeList)
```

Comment:
I personally like to name the variable instead of a single letter, but it's as you wish :)

Reply:
I don't yet know all the idioms, this one is definitely a Go one, thank you for letting me know about this. :)